### PR TITLE
To define success for the project, created outline of features for mvp.

### PR DESCRIPTION
**Pull request by johana-star**

### CODE REVIEW

File: projects/social_network.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+I can see which members have made a friend request
+
+As a member
+I can approve a friend request
+
+As a member
+I can deny a friend request
+
+As a member
+I cannot see if someone denied my friend request
```
Comment (zspencer) [inline projects/social_network.md]: Does this mean that they just see the request is still outstanding, or does it mean they can resubmit a new friend request?

File: projects/social_network.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+I can post a picture to my feed
+
+As a member
+I can see a feed of my friends' activity
+
+As a friend of another member
+I can see their feed
+
+As an admin
+I can ban members from logging into the site
```
Comment (zspencer) [inline projects/social_network.md]: Admin features seem out of band for an MVP.

File: projects/social_network.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+I cannot see if someone denied my friend request
+
+As a member
+I can add a post to my feed
+
+As a member
+I can post a picture to my feed
+
+As a member
+I can see a feed of my friends' activity
```
Comment (zspencer) [inline projects/social_network.md]: I would clarify if this is an aggregate feed of all friends activity, or if I can just see activity by visiting their profile.

An Aggregate Feed is cooler, but More Work. Alternatively, we could use this as a learning moment about how a programmers duty is to do the simplest thing that can possibly work, not jump to the "fun but hard" problem.

**Comment by zspencer** [comment]
Looks pretty good. The only thing I would request you change before I merge is that admin seems like more than MVP level stuff.

**Comment by johana-star** [comment]
I've incorporated changes. Please review them and if they pass the sniff test, merge!

File: projects/social_network.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+==== Visitors ====
+
+* Visitor cannot view any members content.
+* Visitor can see example content from the site.
+* Visitor can sign up for an account.
+
+==== Members ====
+
+* Member can request other members become my friend.
+* Member can see which members have made a friend request.
```
Comment (zspencer) [inline projects/social_network.md:7]: I'm not sure what this means? Is it they can see which members they've invited to be their friend?

File: projects/social_network.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+* Member can approve a friend request.
+* Member can deny a friend request.
+* Member cannot see if someone denied my friend request.
+* Member does not see friend requests they denied.
+* Member can add a post to my feed.
```
Comment (zspencer) [inline projects/social_network.md]: What feed is my feed? Is it "their" feed?

File: projects/social_network.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+* Member can post a picture to my feed.
```
Comment (zspencer) [inline projects/social_network.md]: Picture seems out of scope for an MVP

File: projects/social_network.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+==== Friends ====
+
+* A friend of another member can see their feed.
+* Member cannot see another member's feed unless they are a friend.
+
+==== Aggregate stream (might be a challenge) ====
+
+* Member can see an aggregate feed of my friends' activity.
```
Comment (zspencer) [inline projects/social_network.md:18]: This also seems out of scope for an MVP. an MVP is literally "Hey! Here's my friends! I CAN SEE WHAT THEY DONE DID!"

File: projects/social_network.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+== Advanced Features ==
```
Comment (zspencer) [inline projects/social_network.md]: I wouldn't call these "advanced" but I would call them "Additional"

We want to give a _huge_ list of features so boots have a ton of stuff to do and cannot actually complete everything within a week.

"Advanced" features would be stuff that requires additional gems, or a more advanced object model.

**Comment by zspencer** [comment]
Getting nicer.

One thing I'd recommend is to place the features in an ordered backlog, as opposed to grouping them by actor. Boots can, and should, be choosing the final order, but grouping them makes it somewhat ambiguous.

**Comment by johana-star** [comment]
This is closer. Please nitpick!

File: projects/social_network.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+* Member can add a post to their feed.
+* Member can request other members become my friend.
+* Member can see which members have made a friend request.
+* Member cannot see another member's feed unless they are a friend.
+* Member can approve a friend request.
+* A friend of another member can see their feed.
+* Member can deny a friend request.
+* Member cannot see if someone denied their friend request.
+* Member does not see friend requests they denied.
+* Visitor can see example content from the site.
```
Comment (zspencer) [inline projects/social_network.md:17]: Seeing example content even seems outside of scope of the MVP. However, this looks much more succinct than before.

File: projects/social_network.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
+== Minimum Viable Product ==
```
Comment (zspencer) [inline projects/social_network.md]: Any reason for the `==` vs `##`?

File: projects/social_network.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+* Visitor cannot view any members content.
+* Visitor can sign up for an account.
+* Member can add a post to their feed.
+* Member can request other members become my friend.
+* Member can see which members have made a friend request.
+* Member cannot see another member's feed unless they are a friend.
```
Comment (zspencer) [inline projects/social_network.md:9]: I don't see a feature for "Member can see another member's feed"

File: projects/social_network.md [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+* Member can approve a friend request.
+* A friend of another member can see their feed.
+* Member can deny a friend request.
```
Comment (zspencer) [inline projects/social_network.md:14]: Denying a friend request seems like you could wait until after the MVP. Sure, it's annoying to have the request still sitting there, but it's not _mandatory_

**Comment by zspencer** [comment]
Part of the goal is that boots have a product pushed to heroku within the first couple days. Anything we add to the MVP gets in the way of their pushing to production. This is why I'm so OCD about what is in the MVP vs. what is in Additional Features.

**Comment by zspencer** [comment]
Exxxcelllent! Now we need another branch to make the quantity of features overwhelming, and one for the "here are advanced features that will require you to learn a new gem or 7"
